21IanuarieHW/src/StoreProblem.java

```diff
@@ -0,0 +1,83 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.List;
+
+class Product {
+    float price;
+    String validDate;
+    float weight;
+}
+
+class AnimalProduct extends Product {
+    String animProdType;
+    float storageTemp;
+}
+
+class VegetalProduct extends Product {
+    String vegProdType;
+    String vitamins;
+}
+
+public class StoreProblem {
+
+    public static void main(String[] args) {
+        List stock = new ArrayList();
+        System.out.println("1. Create product and add it to stock");
+        System.out.println("2. Sell product");
+        System.out.println("3. Display daily sales report");
+        System.out.println("4. Exit");
+
+        Scanner keyboard = new Scanner(System.in);
+        int choice1 = keyboard.nextInt();
+        //create product and add to stock
+        if (choice1 == 1) {
+            System.out.println("Product type (1 - animal; 2 - vegetable):");
+            int choice2 = keyboard.nextInt();
+            if (choice2 == 1) {
+                System.out.println("Please insert the price, validity date, weight, type and " +
+                        "storage temperature for the animal product");
+                AnimalProduct prodAnimal = new AnimalProduct();
+                prodAnimal.price = keyboard.nextFloat();
+                prodAnimal.validDate = keyboard.nextLine();
+                prodAnimal.weight = keyboard.nextFloat();
+                prodAnimal.animProdType = keyboard.nextLine();
+                prodAnimal.storageTemp = keyboard.nextFloat();
+
+                //pot adauga asa in lista stock un obiect?
+                stock.add(prodAnimal);
```

Comment:
yes you can, and if you specified that the list is of product you can still add an animal product due to polimorphism